File: app/src/main/java/com/sanislo/movieapp/presentation/MainActivity.java
package com.sanislo.movieapp.presentation;

import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.sanislo.movieapp.R;
import com.sanislo.movieapp.presentation.upcomingMovies.UpcomingMoviesFragment;

import javax.inject.Inject;

import dagger.android.AndroidInjection;
import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.HasSupportFragmentInjector;

public class MainActivity extends AppCompatActivity
        implements HasSupportFragmentInjector, HasDualPaneSupport {

    @Inject
    DispatchingAndroidInjector<Fragment> fragmentDispatchingAndroidInjector;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        if (savedInstanceState != null) return;
        getSupportFragmentManager().beginTransaction()
                .add(R.id.fl_movie_list_container, UpcomingMoviesFragment.newInstance())
                .commit();
    }

    @Override
    public AndroidInjector<Fragment> supportFragmentInjector() {
        return fragmentDispatchingAndroidInjector;
    }

    @Override
    public boolean isInDualPaneMode() {
        return findViewById(R.id.fl_movie_details_container) != null;
    }

    @Override
    public int getLeftContainerId() {
        return R.id.fl_movie_list_container;
    }

    @Override
    public int getRightContainer() {
        return R.id.fl_movie_details_container;
    }
}

File: app/src/main/java/com/sanislo/movieapp/presentation/MainActivity.kt
package com.sanislo.movieapp.presentation

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.sanislo.movieapp.R
import com.sanislo.movieapp.presentation.upcomingMovies.UpcomingMoviesFragment
import dagger.android.AndroidInjection
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.support.HasSupportFragmentInjector
import javax.inject.Inject

class MainActivity : AppCompatActivity(), HasSupportFragmentInjector, HasDualPaneSupport {

    @Inject
    lateinit var fragmentDispatchingAndroidInjector: DispatchingAndroidInjector<Fragment>

    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (savedInstanceState != null) return
        supportFragmentManager.beginTransaction()
                .add(R.id.fl_movie_list_container, UpcomingMoviesFragment.newInstance())
                .commit()
    }

    override fun supportFragmentInjector(): AndroidInjector<Fragment>? {
        return fragmentDispatchingAndroidInjector
    }

    override fun isInDualPaneMode(): Boolean {
        return findViewById<View>(R.id.fl_movie_details_container) != null
    }

    override fun leftContainerId(): Int {
        return R.id.fl_movie_list_container
    }

    override fun rightContainer(): Int {
        return R.id.fl_movie_details_container
    }
}
